keras_nlp/models/roberta.py

```diff
@@ -0,0 +1,302 @@
+    ```
+    """
+
+    def __init__(
+        self,
+        vocabulary_size,
+        num_layers,
+        num_heads,
+        hidden_dim,
+        intermediate_dim,
+        dropout=0.1,
+        max_sequence_length=512,
+        bos_token_index=0,
+        name=None,
+        trainable=True,
+    ):
+
+        token_id_input = keras.Input(
+            shape=(None,), dtype="int32", name="input_ids"
+        )
+        input_mask = keras.Input(
+            shape=(None,), dtype="int32", name="input_mask"
+        )
+
+        # Embed tokens and positions.
+        token_and_position_embedding_layer = TokenAndPositionEmbedding(
+            vocabulary_size=vocabulary_size,
+            sequence_length=max_sequence_length,
+            embedding_dim=hidden_dim,
+            embeddings_initializer=_roberta_kernel_initializer(),
+            name="token_and_position_embeddings",
+        )
+        embedding = token_and_position_embedding_layer(token_id_input)
+
+        # Sum, normailze and apply dropout to embeddings.
+        x = keras.layers.LayerNormalization(
+            name="embeddings_layer_norm",
+            axis=-1,
+            epsilon=1e-5,  # Original paper uses this epsilon value
+            dtype=tf.float32,
+        )(embedding)
+        x = keras.layers.Dropout(
+            dropout,
+            name="embeddings_dropout",
+        )(x)
+
+        # Apply successive transformer encoder blocks.
+        for i in range(num_layers):
+            x = TransformerEncoder(
+                num_heads=num_heads,
+                intermediate_dim=intermediate_dim,
+                activation=lambda x: keras.activations.gelu(
+                    x, approximate=True
+                ),
+                dropout=dropout,
+                kernel_initializer=_roberta_kernel_initializer(),
+                name=f"""transformer_layer_{i}""",
+            )(x, padding_mask=input_mask)
+
+        # Construct the two RoBerta outputs.
+        output = x
+
+        # Instantiate using Functional API Model constructor
+        super().__init__(
+            inputs={
+                "input_ids": token_id_input,
+                # "segment_ids": segment_id_input,
+                "input_mask": input_mask,
+            },
+            outputs=output,
+            name=name,
+            trainable=trainable,
+        )
+        # All references to `self` below this line
+        self.token_embedding = (
+            token_and_position_embedding_layer.token_embedding
+        )
+        self.vocabulary_size = vocabulary_size
+        self.hidden_dim = hidden_dim
+        self.intermediate_dim = intermediate_dim
+        self.num_layers = num_layers
+        self.num_heads = num_heads
+        self.max_sequence_length = max_sequence_length
+        self.intermediate_dim = intermediate_dim
+        self.dropout = dropout
+        # BOS token '<s>' is equivalent to '[CLS]' from BERT
+        self.bos_token_index = bos_token_index
```

Comment:
Should we try for uniformity here? `some_bert.start_token_index` and `some_roberta.start_token_index`? Or do we think users will thing in terms of CLS when using BERT and BOS when using RoBERTa